core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -830,9 +835,11 @@ class LogManager(logDirs: Seq[File],
         cleaner.abortCleaning(topicPartition)
         cleaner.updateCheckpoints(removedLog.dir.getParentFile)
       }
-      removedLog.renameDir(Log.logDeleteDirName(topicPartition))
+
       checkpointLogRecoveryOffsetsInDir(removedLog.dir.getParentFile)
       checkpointLogStartOffsetsInDir(removedLog.dir.getParentFile)
+      removedLog.close()
+      removedLog.renameDir(Log.logDeleteDirName(topicPartition))
```

Comment:
Have we verified that it's safe to do the rename after the checkpoint calls? Also, it seems like the following code in `replaceCurrentWithFutureLog` does the `rename` before `close` and needs to be updated too.

```scala
        sourceLog.renameDir(Log.logDeleteDirName(topicPartition))
        // Now that replica in source log directory has been successfully renamed for deletion.
        // Close the log, update checkpoint files, and enqueue this log to be deleted.
        sourceLog.close()
        checkpointLogRecoveryOffsetsInDir(sourceLog.dir.getParentFile)
        checkpointLogStartOffsetsInDir(sourceLog.dir.getParentFile)
```